src/Task/domain/entity/Task.ts

```diff
@@ -1,30 +1,115 @@
-import { TaskInterface } from './TaskInterface';
+import {
+  TaskInterface,
+  User,
+  TimeSlot,
+  SchedulingOption,
+} from './TaskInterface';
 
 export class Task {
   taskId?: string;
   title: string;
   description: string;
-  user: {
-    userId: string;
-    fullName: string;
-  };
+  user: User;
+  category: string;
+  location?: string;
+  budget: number;
+  scheduling: SchedulingOption;
+  specificDate?: Date;
+  timeslot?: TimeSlot;
+  createdon?: Date;
 
   constructor(data: TaskInterface) {
     this.taskId = data.taskId;
     this.title = data.title;
     this.description = data.description;
     this.user = data.user;
+    this.category = data.category;
+    this.location = data.location;
+    this.budget = data.budget;
+    this.scheduling = data.scheduling;
+    this.specificDate = data.specificDate;
+    this.timeslot = data.timeslot;
+    this.createdon = data.createdon;
   }
 
-  taskIsValid?(): { valid: boolean; error?: string } {
-    if (!this.title || !this.description) {
-      return { valid: false, error: 'Empty or invalid title or description' };
+  private errorObject?(
+    field: string,
+    error: string,
+  ): { field: string; error: string } {
+    return { field, error };
+  }
+
+  taskIsValid?(): {
```

Comment:
don't need to use option(?) here

Reply:
I have problem at test when I create a task, I also need to create taskIsValid method as well. :(
So to avoid that, I set as optional for that method. 
If I set required then I need to solve it test cases.
Help me brainstorm it :)